scripts.py

```diff
@@ -264,6 +291,67 @@ def run(self, data_source_name, metadata_records, epub_directory, cover_director
             self._db.commit()
 
 
+class OPDSImportScript(BaseOPDSImportScript):
+
+    """An OPDSImportScript class that finds a collection based on its
+    DataSource instead of parsing the command line.
+    """
+
+    IMPORTER_CLASS = None
+
+    def __init__(self, importer_class, data_source_name,
+                 collection_data=None, _db=None
+    ):
+        super(OPDSImportScript, self).__init__(_db=_db)
+
+        self.IMPORTER_CLASS = importer_class
+
+        # Create Collection(s) for this import.
+        collection_data = collection_data or importer_class.collection_data()
+        self.create_collections(data_source_name, collection_data)
+
+        # Find every Collection with this DataSource.
+        collections = Collection.by_datasource(self._db, data_source_name)
+        self.collections = collections.all()
+
+    def create_collections(self, data_source_name, collection_data):
```

Comment:
I found `collection_data` difficult to understand when I first saw it above, but I don't know if it's worth changing.

Basically this is a list of arguments that are used as a shortcut to configure a Collection with different settings. Ideally the list would be a list of argument dictionaries each of which would be passed as **kwargs into some method of Collection.

But it really doesn't matter that much, certainly not enough to go in and add a method to Collection. So I think it's fine to leave it alone.